### PR TITLE
quic tile buffer overrun fixes

### PR DESCRIPTION
**Pull request by nbridge-jump**

### CODE REVIEW

File: src/app/fdctl/topology.c [hunk 1]
```diff
@@ -184,7 +184,7 @@ fd_topo_workspace_fill( fd_topo_t *      topo,
       case FD_TOPO_LINK_KIND_QUIC_TO_VERIFY:
         /* The QUIC tile stashes some information in the dcache app region, this
             should probably be changed. */
-        dcache_app_sz = fd_quic_dcache_app_footprint( link->depth );
+        dcache_app_sz = fd_quic_dcache_app_footprint( ( link->depth + link->burst ) * FD_CHUNK_SZ );
```
Comment (mmcgee-jump) [inline src/app/fdctl/topology.c:187]: I don't understand why you multiply by FD_CHUNK_SZ here?

**Review by ripatel-fd — DISMISSED** [review]

File: src/app/fdctl/topology.c (hunk 1 above)
Comment (mmcgee-jump) [reply src/app/fdctl/topology.c:187]: I don't think that's right, the dcache_app_footprint function already does that multiplication internally.

Comment (nbridge-jump) [reply src/app/fdctl/topology.c]: I'm not sure why I thought it was necessary. I've removed it, and it doesn't overflow in tests so far.

File: src/app/fdctl/run/tiles/fd_quic.c [hunk 2]
```diff
@@ -277,8 +280,16 @@ fd_quic_chunk_idx( ulong chunk0,
 FD_FN_CONST static inline fd_quic_msg_ctx_t *
 fd_quic_dcache_msg_ctx( uchar * app_laddr,
                         ulong   chunk0,
-                        ulong   chunk ) {
+                        ulong   chunk,
+                        ulong   depth,
+   ulong wmark ) {
   fd_quic_msg_ctx_t * msg_arr = (fd_quic_msg_ctx_t *)app_laddr;
+  ulong idx = fd_quic_chunk_idx( chunk0, chunk );
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_quic.c:287]: :shrug: Let's remove these debugging additions now that it's fixed?

File: src/app/fdctl/run/tiles/fd_quic.c [hunk 3]
```diff
@@ -264,7 +266,8 @@ mux_ctx( void * scratch ) {
 FD_FN_CONST static inline ulong
 fd_quic_chunk_idx( ulong chunk0,
                    ulong chunk ) {
-  return ((chunk-chunk0)*FD_CHUNK_FOOTPRINT) / fd_ulong_align_up( FD_TPU_DCACHE_MTU, FD_CHUNK_FOOTPRINT );
+  ulong j = (chunk-chunk0) * FD_CHUNK_SZ / fd_ulong_align_up( FD_TPU_DCACHE_MTU, 2*FD_CHUNK_SZ );
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_quic.c:269]:
```suggestion
  return (chunk-chunk0) * FD_CHUNK_SZ / fd_ulong_align_up( FD_TPU_DCACHE_MTU, 2*FD_CHUNK_SZ );
```

File: src/tango/quic/fd_quic.c [hunk 4]
```diff
@@ -127,6 +133,13 @@ fd_quic_footprint_ext( fd_quic_limits_t const * limits,
   if( FD_UNLIKELY( !tls_footprint ) ) { FD_LOG_WARNING(( "invalid fd_quic_tls_footprint" )); return 0UL; }
   offs                += tls_footprint;
 
+  /* allocate space for fd_ip_t */
```
Comment (mmcgee-jump) [inline src/tango/quic/fd_quic.c:136]: How did this come back? I moved it to be passed in by the caller, it makes the seccomp logic cleaner.

File: src/tango/quic/fd_quic.h [hunk 5]
```diff
@@ -195,6 +195,9 @@ struct __attribute__((aligned(16UL))) fd_quic_limits {
 
   ulong  tx_buf_sz;        /* per-stream, tx buf sz in bytes          */
   /* the user consumes rx directly from the network buffer */
+
+  ulong  arp_entries;      /* instance-wide, space for arp entries */
```
Comment (mmcgee-jump) [inline src/tango/quic/fd_quic.h:199]: Also why is this coming back? It's configured by the caller.

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
The specific fix looks OK, but why are there ARP decls / changes in here?

**Comment by ripatel-fd** [comment]
Hopefully I can rewrite this entire component in the next few hours.
With CBMC proofs that it's free of memory issues and data races. 
Apologies for this original garbage code, I feel embarrassed about it.

**Comment by ripatel-fd** [comment]
@nbridge-jump I'm closing this since the faulty dcache code has been replaced entirely in https://github.com/firedancer-io/firedancer/pull/945

**Pull request closed without merging**